### DISCUSSION TITLE
OS Ubuntu, RB Miner does not recognize 2nd CPU

### DISCUSSION THREAD

**Discussion by skyguynca** [RainbowMiner/RainbowMiner#1672] [Q&A] [answered]
I am running a server MB, 2 Intel E5-2650v1 cpu's with Ubuntu Server 20.04 LTS. It is only finding one cpu, in Ubuntu when I check with top, htop or lscpu it shows both cpu's. How do I fix that?

**Comment by RainbowMiner** [comment, marked as answer]
Linux handles multiple CPUs as one, internally. The different cores will be addressed via affinity settings. Normally, RainbowMiner should address all full cores and set the correct number of threads automatically. 

Since your Intel CPUs have 8 cores/16 threads each, RainbowMiner should default to the following settings (check this via "Running Config" page on `http://localhost:4000`)
```
  "CPUMiningAffinity": "0x55555555",
  "CPUMiningThreads": "16",
```

You can tune this up a bit by editing the config.txt:
```
  "CPUMiningAffinity": "0xFFFFFFF5",
  "CPUMiningThreads": "30",
```

The first 4 hex numbers address CPU1 and the second 4 hex numbers address CPU0.
